src/coreclr/src/vm/eventpipe.cpp

```diff
@@ -435,6 +450,27 @@ void EventPipe::Disable(EventPipeSessionID id)
     }
     CONTRACTL_END;
 
+    // EventPipe::Disable is called synchronously since the diagnostics server is
+    // single threaded.  HOWEVER, if the runtime was suspended in EEStartupHelper,
+    // then EventPipe::FinishInitialize might not have executed yet.  Disabling a session
+    // needs to either happen before we resume or after initialization.  We briefly take the
+    // lock to check s_CanStartThreads to check whether we've finished initialization.  We 
+    // also check whether we are still suspended in which case we can safely disable the session
+    // without deferral.
+    {
+        CrstHolder _crst(GetLock());
+        if (!s_CanStartThreads && !IpcStreamFactory::AnySuspendedPorts())
+        {
+            s_rgDeferredDisableEventPipeSessionIds.Push(id);
+            return;
+        }
+    }
+
+    DisableHelper(id);
```

Comment:
@sywhang can you double check my logic here.  We can chat on Monday about this case.

Reply:
I am OOF until Tuesday but I can take a look at the changes over the weekend